pkgs/servers/monitoring/karma/generate-dependencies.sh

```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p nodePackages.node2nix nodejs-18_x gnused wget
+
+# Usage: ./generate-dependencies.sh <version>
+# Use the same version format as in ./default.nix (without the 'v')
+
+set -e
+
+version=$1
+
+
+echo "Karma version: $version"
+cd "$(dirname $(readlink -f $0))"
+
+wget -O ./package-lock.json https://raw.githubusercontent.com/prymitive/karma/v$version/ui/package-lock.json
+wget -O ./package.json https://raw.githubusercontent.com/prymitive/karma/v$version/ui/package.json
+
+sed -i -e 's/"name": "ui",/"name": "karma-ui",/g' ./package.json
+sed -i -e 's/"name": "ui",/"name": "karma-ui",/g' ./package-lock.json
+
+node2nix --development -l ./package-lock.json -i ./package.json -o ./node-packages.nix -c ./node-composition.nix -e ./node-env.nix
+
+sed -i -e 's/nodejs-12_x/nodejs-18_x/g' ./node-composition.nix
```

Comment:
This must be supplied as an arg to node2nix.

Reply:
nodejs-18 support hasn't made it to release yet, the latest stable version (1.11) only supports versions through nodejs-16.  Should I try building the deps with a development version of node2nix or wait for 1.12 to release?

Reply:
Can we fetchpatch https://github.com/svanderburg/node2nix/commit/d0384d77543ffe42b572f70cc21d8ee3324b7183 into nixpkgs node2nix?

Reply:
Should I include it in this PR or create a new one for that?

Reply:
if we do it in another PR we can merge it really fast and get it out of the way while we finish this one.

Reply:
Ah, we're in luck, somebody pushed this 4 days ago https://github.com/NixOS/nixpkgs/commit/07b207c5e9a47b640fe30861c9eedf419c38dce0#diff-8713b31dda472d6b3b0d2129c99c7d5eb494a0e438d26f1c853750b1ac515c19

Reply:
I'll build the latest node2nix and generate the deps from there.

Reply:
@SuperSandro2000 https://github.com/NixOS/nixpkgs/pull/191922/files#diff-b81d157ce4e259ad822bcc22b41f72aa8c575a446ba0477512fac88d71a063c2R22 works with the master version of node2nix